src/index.js

```diff
@@ -133,3 +126,171 @@ export class WealthscopeSdk {
     }
   }
 }
+
+export class WealthscopeApiClient {
+  /**
+   * Constructor for the Wealthscope Client API.
+   * Stores the authorization token for the current session, and the base URl.
+   * @param {object} opts 
+   */
+  constructor(opts) {
+    this.token = null;
+
+    this.opts = Object.assign(
+        {
+          wealthscopeUrl: 'https://api.staging-bus.wealthscope.ca/v1',
+        },
+        opts
+    );
+  }
+
+  /**
+   * Function that returns the current version from the package.json.
+   * @returns {string}
+   */
+  static version() {
+    return version;
+  }
+
+  /**
+   * This retrieves a valid token and sets this.token to be the valid token.
+   * @param {string} jwtData 
+   * @returns {Promise}
+   */ 
+  login(jwtData) {
+    return fetch(this.opts.wealthscopeUrl + '/auth/authenticate/', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({token: jwtData})
+    })
+        .then((response) => response.json())
+        .then(({token}) => {
+          this.token = token;
+        })
+        .catch((err) => {
+          // eslint-disable-next-line no-console
+          console.log('ERROR REQUESTING TOKEN: ', err);
+        });
+  }
+
+  /**
+   * This resets the token to null.
+   */
+  logout() {
+    this.token = null;
+  }
+
+  /**
+   * Fetch call returning a Promise, requires a URL rout and a vailid token.
+   * @param {string} url 
+   * @returns {Promise}
+   */
+  get(url) {
+    const postOptions = this._getFetchOptions('GET');
+    return fetch(this._constructUrl(url), postOptions)
+      .then(response => response.json())
+      .catch(err => console.log(err));
+  }
+
+  /**
+   * Fetch call returning a Promise, requires a URL rout, a body and a valid
```

Comment:
route*
also, it doesn't require a valid token as a parameter. this is a bit ambiguous. can we clear it up? maybe say that `login` needs to be called successfully before we call this method?
why is it on `put` and not on `get` ?